Frontend/CO.CDP.OrganisationApp/ThirdPartyApiClients/CompaniesHouse/CompaniesHouseApi.cs

```diff
@@ -46,14 +51,24 @@ public class CompaniesHouseApi(IConfiguration configuration,
                 .GetAsync()
                 .ReceiveJson<CompanyProfile>();
         }
+        catch (FlurlHttpException ex) when ((ex.StatusCode == (int)HttpStatusCode.NotFound) || (ex.StatusCode == (int)HttpStatusCode.InternalServerError))
+        {
+        }
```

Comment:
Aren't we swallowing 500 errors this way?

Reply:
Yes, but if we don't do this we'll end up with the same errors in the logs whenever an incorrect company number is entered during registration.

Reply:
If there's no way to distinguish between the two types of 500 errors (invalid number vs an actual problem with the api), then I think we'll need to log all 500 errors :(

Otherwise, we'll miss the cases when there's an actual problem with the API.

Reply:
Yeah, this was the dilemma I faced when I developed it, which is why I thought it wise to validate the company number, but Sammy would prefer us not to do it.

Reply:
Ok, I've excluded 404s as these aren't really an error.